src/core_pdf/impl/engine/spec/s_07_objects/resolver.py

```
@@ -308,3 +322,28 @@ def recover_indirect_object(self, lexer: PdfLexer, offset: int) -> object:
+            search_pos = marker + 3
+            parsed = parse_object_marker_prefix(data, marker)
+            if parsed is None:
+                continue
+            offset, object_number, generation_number = parsed
+            if object_number != ref.object_number or generation_number != ref.generation_number:
+                continue
+            lexer.rewind(offset)
+            try:
+                recovered = lexer.parse_indirect_object()
+            except Exception:
+                continue
```

Ruff (0.16.1)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)